src/node/node.rs

```diff
@@ -92,7 +104,46 @@ impl Node {
         (sink_halves, stream_halves)
     }
 
-    async fn run_operators(&mut self) -> Result<(), String> {
+    /// Splits a vector of TCPStreams into `ControlMessageHandler`, `ControlSender`s and `ControlReceiver`s.
+    async fn split_control_streams(
+        &self,
+        streams: Vec<(NodeId, TcpStream)>,
+    ) -> (
+        ControlMessageHandler,
+        Vec<ControlSender>,
+        Vec<ControlReceiver>,
+    ) {
+        let mut control_receivers = Vec::new();
+        let mut control_senders = Vec::new();
+        let (handler_tx, handler_rx) = mpsc::unbounded_channel();
```

Comment:
What is the rationale of using a `std::mpsc` and not a `tokio::mpsc`? `recv` may block the runtime thread while waiting for a message on the channel.